docs/source/examples/pt_zne.md

```diff
@@ -341,50 +354,65 @@ ideal_value = execute(circuit, noise_level=0.0)
 noisy_value = execute(circuit, noise_level=NOISE_LEVEL)
 
 NUM_TWIRLED_VARIANTS = 300
-twirled_circuits = generate_pauli_twirl_variants(circuit, num_circuits=NUM_TWIRLED_VARIANTS)
+twirled_circuits = generate_pauli_twirl_variants(
+    circuit, num_circuits=NUM_TWIRLED_VARIANTS
+)
 
 # Average results executed over twirled circuits
 from functools import partial
 from mitiq import Executor
-pt_vals = Executor(partial(execute, noise_level=NOISE_LEVEL)).evaluate(twirled_circuits)
+
+pt_vals = Executor(partial(execute, noise_level=NOISE_LEVEL)).evaluate(
+    twirled_circuits
+)
 twirled_result = np.average(pt_vals)
 
 
 print(f"Error without twirling: {abs(ideal_value - noisy_value) :.3}")
 print(f"Error with twirling: {abs(ideal_value - twirled_result) :.3}")
 ```
 
-It is worth noting that Pauli twirling's goal is to only tailor the noise from coherent to incoherent. 
+It is worth noting that Pauli twirling's goal is to only tailor the noise from coherent to incoherent.
 
 Depending on the noise strength, type of coherent noise etc. this transformation might not give better results after the Pauli twirled circuit is executed. See the plot in the [next section](#combining-pauli-twirling-with-zne) for an example.
 
-
 ## Combining Pauli Twirling with ZNE
 
-```{code-cell} ipython3
-from mitiq.zne import execute_with_zne
+To combine Pauli twirling with ZNE, we'll first generate the noise-scaled circuits with `mitiq.zne.construct_circuits`, then apply twirling.
 
-executor=partial(execute, noise_level=NOISE_LEVEL)
+```{code-cell} ipython3
+from mitiq import zne
 
-def combine_zne_pt(list_circuits, executor) -> list[float]:
-    zne_pt_vals = []
+scale_factors = [1, 3, 5]
 
-    for twirled_circuit in list_circuits:
-        zne_pt_vals.append(execute_with_zne(twirled_circuit, executor))
+noise_scaled_circuits = zne.construct_circuits(circuit, scale_factors)
+noisy_executor = partial(execute, noise_level=NOISE_LEVEL)
+noise_scaled_expvals = []
+for noise_scaled_circuit in noise_scaled_circuits:
+    pt_variants = generate_pauli_twirl_variants(
+        noise_scaled_circuit, num_circuits=NUM_TWIRLED_VARIANTS
+    )
+    noise_scaled_expvals.append(
+        sum(noisy_executor(pt_variant) for pt_variant in pt_variants)
+        / NUM_TWIRLED_VARIANTS
+    )
```

Comment:
why didn't you use `np.average` here? You can get the Pauli twirled values through evaluate, and all that's left then is to average them. It might be easier to read the code this way. Feel free to accept or reject this suggestion.

Reply:
I think I had tried this, but `np.average` isn't able to compute an average of a generator. That said, switching the executor function to an instance of the `Executor` class made this possible.